### PR TITLE
Refactor online debug fe

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] New feature provided

- Related issues
split from #964 
FE of #835 
Depend on: https://github.com/apache/apisix-dashboard/pull/979  
___

### CODE REVIEW

File: web/src/pages/Route/List.tsx [hunk 1]
```diff
@@ -14,19 +14,21 @@
...
 import ProTable, { ProColumns, ActionType } from '@ant-design/pro-table';
 import { Button, Popconfirm, notification, Tag, Space } from 'antd';
 import { history, useIntl } from 'umi';
-import { PlusOutlined } from '@ant-design/icons';
+import { PlusOutlined, BugOutlined } from '@ant-design/icons';
+import { timestampToLocaleString } from '@/helpers';
 
 import { fetchList, remove } from './service';
-import { timestampToLocaleString } from '@/helpers';
+import { DebugDrawDebugDrawView } from './components/DebugViews';
```
Comment (LiteSun) [inline web/src/pages/Route/List.tsx:26]: weird name 🤔

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { Input, Select, Card, Tabs, Form, Drawer, Spin, notification } from 'antd';
+import { useIntl } from 'umi';
+import CodeMirror from '@uiw/react-codemirror';
+import { PanelSection } from '@api7-dashboard/ui';
+import queryString from 'query-string';
+import Base64 from 'base-64';
```
Comment (LiteSun) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:23]: should be a blank line here.

File: web/src/pages/Route/components/DebugViews/DebugParamsView.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { Form, Input, Row, Col, Checkbox } from 'antd';
+import { useIntl } from 'umi';
+import { MinusCircleOutlined } from '@ant-design/icons';
```
Comment (LiteSun) [inline web/src/pages/Route/components/DebugViews/DebugParamsView.tsx:20]: ditto

File: web/src/pages/Route/components/DebugViews/index.less [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ */
+@import '~antd/es/style/themes/default.less';
+
+.routeDebugDraw {
+  :global {
+    .ant-card-body {
+      padding: 0;
+    }
+    .ant-select-selector {
+      height: 40px !important;
```
Comment (LiteSun) [inline web/src/pages/Route/components/DebugViews/index.less:25]: Be careful with `!important`, is there no better way？🤔

File: web/src/pages/Route/components/DebugViews/index.less [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    .ant-select-selection-item {
+      line-height: 40px !important;
```
Comment (LiteSun) [inline web/src/pages/Route/components/DebugViews/index.less:28]: ditto.

File: web/src/pages/Route/List.tsx (hunk 1 above)
Comment (liuxiran) [reply web/src/pages/Route/List.tsx]: done

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx (hunk 2 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:23]: done

File: web/src/pages/Route/components/DebugViews/DebugParamsView.tsx (hunk 3 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/DebugParamsView.tsx:20]: done

File: web/src/pages/Route/components/DebugViews/index.less (hunk 4 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/index.less]: good catch~! set `Select size="large"` can solve this problem, many thanks~!

done

File: web/src/pages/Route/components/DebugViews/index.less (hunk 5 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/index.less]: done

**Review by LiteSun — APPROVED** [review]

File: web/src/pages/Route/components/DebugViews/AuthenticationView.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+              defaultValue={authType}
+              onChange={(event) => {
+                const currentValue = event.target.value;
+                setAuthType(currentValue);
+                props.form.setFieldsValue({ autyType: currentValue });
+              }}
+            >
+              <Radio value="none">none</Radio>
+              {AUTH_LIST.map((type) => (
+                <Radio value={type}>{type}</Radio>
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/AuthenticationView.tsx:116]: key is needed

File: web/src/pages/Route/components/DebugViews/AuthenticationView.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+          <Form.Item name="authType">
+            <Radio.Group
+              defaultValue={authType}
+              onChange={(event) => {
+                const currentValue = event.target.value;
+                setAuthType(currentValue);
+                props.form.setFieldsValue({ autyType: currentValue });
+              }}
+            >
+              <Radio value="none">none</Radio>
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/AuthenticationView.tsx:114]: use `None` as label would be better

```suggestion
              <Radio value="none">None</Radio>
```

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+            defaultValue={httpMethod}
+            style={{ width: '20%' }}
+            onChange={(value) => {
+              setHttpMethod(value);
+              setShowBodyTab(!(methodWithoutBody.indexOf(value) > -1));
+            }}
+            size="large"
+          >
+            {HTTP_METHOD_OPTION_LIST.map((method) => {
+              return <Option value={method}>{method}</Option>;
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:173]: key is also needed

File: web/src/pages/Route/components/DebugViews/DebugParamsView.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  const { formatMessage } = useIntl();
+
+  return (
+    <Form name="dynamic_form_nest_item" className={styles.routeDebugDraw} form={props.form}>
+      <Form.List name="params">
+        {(fields, { add, remove }) => {
+          return (
+            <>
+              {fields.map((field, index) => (
+                <Row gutter={16}>
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/DebugParamsView.tsx:34]: key

File: web/src/pages/Route/service.ts [hunk 10]
```diff
@@ -85,3 +85,10 @@ export const checkHostWithSSL = (hosts: string[]) =>
     method: 'POST',
     data: hosts,
   });
+
+export const debugRoute = (params: RouteModule.debugRequest) => {
+  return request('/debug-request-forwarding', {
+    method: 'post',
+    data: { ...params },
```
Comment (juzhiyuan) [inline web/src/pages/Route/service.ts:92]: may don't need to destruct here?

**Review by juzhiyuan — COMMENTED** [review]
LGTM except some React tips.

**Review by juzhiyuan — APPROVED** [review]
Let's wait for #979

File: web/src/pages/Route/components/DebugViews/AuthenticationView.tsx (hunk 6 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/AuthenticationView.tsx]: done

File: web/src/pages/Route/components/DebugViews/AuthenticationView.tsx (hunk 7 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/AuthenticationView.tsx]: done

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx (hunk 8 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/DebugDrawView.tsx]: done

File: web/src/pages/Route/components/DebugViews/DebugParamsView.tsx (hunk 9 above)
Comment (liuxiran) [reply web/src/pages/Route/components/DebugViews/DebugParamsView.tsx:34]: done

File: web/src/pages/Route/service.ts (hunk 10 above)
Comment (liuxiran) [reply web/src/pages/Route/service.ts]: yes, you are right,already fixed :)

**Comment by liuxiran** [comment]
https://github.com/apache/apisix-dashboard/pull/1026
this issue was recured in current branch, tring to resolve~

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 11]
```diff
@@ -170,7 +172,7 @@ const DebugDrawView: React.FC<RouteModule.DebugDrawProps> = (props) => {
             size="large"
           >
             {HTTP_METHOD_OPTION_LIST.map((method, index) => {
-              return <Option key={`${method}_${index}`} value={method}>{method}</Option>;
+              return <Option key={index.toString()} value={method}>{method}</Option>;
```
Comment (juzhiyuan) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:175]: why not use `method` as key?

File: web/src/pages/Route/components/DebugViews/DebugDrawView.tsx [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+            style={{ width: '20%' }}
+            onChange={(value) => {
+              setHttpMethod(value);
+              setShowBodyTab(!(methodWithoutBody.indexOf(value) > -1));
+            }}
+            size="large"
+          >
+            {HTTP_METHOD_OPTION_LIST.map((method) => {
+              return (
+                <Option key={method} value={method}>
```
Comment (liuxiran) [inline web/src/pages/Route/components/DebugViews/DebugDrawView.tsx:177]: thanks for @juzhiyuan 's reminder, already updated.

**Comment by liuxiran** [comment]
> #1026
> this issue was recured in current branch, tring to resolve~

I re-import the new dependency packages and updated yarn.lock, this issue solved, but as a bad effect, I have to rebase the master branch, and use `force-pushed` to update pr, which disturbed code review, sorry for all reviewers.

**Comment by liuxiran** [comment]
tring to yarn install `Cypress` frist to solve the yarn.lock conflict.

**Comment by liuxiran** [comment]
IMHO this pr is ready, it can be merged after https://github.com/apache/apisix-dashboard/pull/979 ^_^

**Pull request merged**